web_renderer/src/server.ts

```diff
@@ -0,0 +1,53 @@
+import express, { Express, Request, Response } from 'express';
+import { Resolution, Url } from './common';
+import { Session } from './session';
+
+export class Server {
+    private server: Express;
+    private sessions: Map<Url, Session>;
+
+    public constructor() {
+        this.server = express();
+        this.server.use(express.json());
+        this.sessions = new Map();
+
+        this.initRoutes();
+    }
+
+    public listen(port: number): void {
+        this.server.listen(port, () => {
+            console.log(`Listening on ${port}`);
+        });
+    }
+
+    private initRoutes(): void {
+        // TODO: better endpoint name
+        this.server.post("/get_frame", this.get_frame.bind(this));
```

Comment:
I would prefer it if api rendering and creating session was a separate call. The logic when you create session if it does not exists seems a bit hacky